readme.md

```diff
@@ -145,6 +146,34 @@ add2(10) // 12
 
 ```
 
+##Closure
+
+A very simplistic definition - A closure is a way of accessing a variable outside its scope.
+Formally, a closure is a technique for implementing lexically scopped named binding. It is a way of storing a function with an environment. 
+
+A closure is a scope which captures local variables of a function for access even after the execution has moved out of the block in which it is defined.
+ie. they allow referencing a scope after the block in which the variables were declared has finished executing. 
+
+
+```js
+function getTicker () {
+  var tick = 0;
+  function ticker () {
+   tick = tick + 1;
+   return tick;
+   }
+  return ticker;
+ }
+ var tickTock = getTicker();
+ tickTock(); //returns 1
+ tickTock(); //returns 2
+```
+The function getTicker() returns a function(internally called ticker), lets store it in a variabke called tickTock.
+
+Ideally, when the function getTicker finishes execution, its scope, with local variable tick should not be accessible. But, it returns 1, 2, 3.. on calling tickTock(). This simply means that, somewhere it keeps a track of the variable tick. 
+
+Lexical scoping is the reason why it is able to find the value of tick - the private variable of the parent which has finished executing. This value is called a Closure. The stack along with the lexical scope of the function is stored and upon re-execution same stack is restored. 
```

Comment:
I don't think the stack is retained exactly. If I remember right the closure just retains a reference to the parent scope that prevents garbage collection